server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -0,0 +1,270 @@
+                    result.updated(knownHost);
+                }
+                else {
+                    result.unchanged(knownHost);
+                }
+            }
+            context.setResult(result);
+        }
+        catch (Exception e) {
+            log.error("HypervisorUpdateJob encountered a problem.", e);
+            context.setResult(e.getMessage());
+            throw new JobExecutionException(e.getMessage(), e, false);
+        }
+    }
+
+    /**
+     * Creates a {@link JobDetail} that runs this job for the given {@link Owner}.
+     *
+     * @param owner the owner to refresh
+     * @return a {@link JobDetail} that describes the job run
+     */
+    public static JobDetail forOwner(Owner owner, String data, Boolean create, Principal principal) {
+        JobDataMap map = new JobDataMap();
+        map.put(JobStatus.TARGET_TYPE, JobStatus.TargetType.OWNER);
+        map.put(JobStatus.TARGET_ID, owner.getKey());
+        map.put(CREATE, create);
+        map.put(DATA, compress(data));
+        map.put(PRINCIPAL, principal);
+
+        // Not sure if this is the best way to go:
+        // Give each job a UUID to ensure that it is unique
+        JobDetail detail = newJob(HypervisorUpdateJob.class)
+            .withIdentity(prefix + Util.generateUUID())
+            .requestRecovery(true) // recover the job upon restarts
+            .usingJobData(map)
+            .storeDurably(true) // required if we have to postpone the job
+            .build();
+
+        return detail;
+    }
+
+    public static byte[] compress(String text) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            OutputStream out = new DeflaterOutputStream(baos);
+            out.write(text.getBytes("UTF-8"));
+            out.close();
+        }
+        catch (IOException e) {
+            throw new AssertionError(e);
+        }
+        return baos.toByteArray();
+    }
+
+    public static String decompress(byte[] bytes) {
+        InputStream in = new InflaterInputStream(new ByteArrayInputStream(bytes));
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            byte[] buffer = new byte[8192];
+            int len;
+            while ((len = in.read(buffer)) > 0) {
+                baos.write(buffer, 0, len);
+            }
+            return new String(baos.toByteArray(), "UTF-8");
+        }
+        catch (IOException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    /*
+     * Create a new hypervisor type consumer to represent the incoming hypervisorId
+     */
+    private Consumer createConsumerForHypervisorId(String incHypervisorId,
+            Owner owner, Principal principal) {
+        Consumer consumer = new Consumer();
+        consumer.setName(incHypervisorId);
+        consumer.setType(new ConsumerType(ConsumerTypeEnum.HYPERVISOR));
+        consumer.setFact("uname.machine", "x86_64");
```

Comment:
This was how it was done in the original. If we don't need or want this I will remove from both. Perhaps it is a placeholder or initialization for the facts.